### ISSUE TITLE
[Feature Request] Add ability to specify own error handler

### ISSUE THREAD

**Issue by MZPL** [Hexagon/croner#154]
Hello.
I'm a new user of this library, and I'm looking forward to migrate from node-cron to croner. I've noticed there's no way to catch and log errors besides putting the whole code in the `try..catch`. This works, but it's not really convenient. 
I looked through the README and found `catch: true` option, but unfortunately, it only catches any potential errors and suppresses them. My suggestion would be to allow passing a function to the `catch` property, which would then be called if an error throws, e.g `{ catch: (err) => console.error(err) }`.

**Comment by Hexagon** [comment]
Will have a look at that 👍

**Comment by Hexagon** [comment]
Released in `5.4.0` - give it a go!

FTR:

   * Tests added
   * Async support

Example, using async function, on Deno:

```javascript
import Cron from "https://deno.land/x/croner@5.4.0/dist/croner.min.mjs";

const onError = async (e) => console.error('Oh shit - ', e);

Cron("* * * * * *", { catch: onError },
	async () => {
		console.log("This will run every second.");
		throw new Error("t");
	}
);
```

**Comment by MZPL** [comment]
Thanks, I didn't expect to see it released that quickly!
I'm using TypeScript and I noticed a small typing issue:
```ts
cron("* * * * * *", () => {
    throw new Error("err!");
  },
  {
    // TS7006: Parameter 'e' implicitly has an 'any' type. (appears if strict mode is enabled)
    // Can be fixed by adding `e: any`
    catch(e) {
      console.error("Error!", e);
    },
  }
);
```
It'd be beneficial for TS users to define the arguments explicitly, so the compiler knows there's an `error` parameter:
`catch?: boolean | ((error: unknown) => void);`

**Comment by Hexagon** [comment]
Thanks for reporting (again)! Will have a look at that for 5.4.1.

**Comment by Hexagon** [comment]
(Hopefully) resolved in `5.4.1` which will be released any minute 👍

**Comment by MZPL** [comment]
> (Hopefully) resolved in `5.4.1` which will be released any minute +1

Yep, I can confirm it's fixed. Thanks!
